File: 242-valid-anagram/valid-anagram.py
from collections import Counter
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        if len(s) != len(t):
            return False
        
        sc = Counter(s)
        tc = Counter(t)

        for ch in sc:
            if sc[ch] != tc[ch]:
                return False
        return True

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {

        if (s.length() != t.length()) {
            return false;
        }

        char[] sArray = s.toCharArray();
        char[] tArray = t.toCharArray();

        Arrays.sort(sArray);
        Arrays.sort(tArray);
        
        

        int i = 0, j= 0;
        while (i < sArray.length && j < tArray.length){
            if (sArray[i] != tArray[j]){
                return false;
            }
            i++;
            j++;
        }
        return true;
        
    }
}